proposals/Interfaces.md

```diff
@@ -0,0 +1,114 @@
+## Interfaces
+
+Interfaces will provide one of the basic type building blocks in the project.
+
+### Basic Definition
+
+An `interface` is a type which provides an abstracted interface (hence, the name) for working with
+one or more concrete types (typically [classes](Classes.md)).
+
+### Defining an interface
+
+Interfaces are defined with the `interface` keyword and represent the accessible members of a `class`:
+
+```
+interface ISomeInterface {
+  function<void> DoSomething()
+  var<string> SomeValue
+  property<bool> SomeProperty { get; set; }
+}
+```
+
+**Note:** We recommend a style of prefixing all interfaces with the letter `I` to distinguish from
+classes.
+
+
+#### Interface Members
+
+Interfaces can contain all the same members as a [class](Classes.md). If an interface member is
+marked as `static`, then it operates on the interface itself, rather than an instance of a particular
+type. Variables **cannot** be marked as static in an interface.
+
+
+#### Interface operators
+
+As operators are defined as always being `static`, an `operator` when declared on an interface
```

Comment:
I'm not 100% sure we need operators on interfaces, but they may provide some nice-to-have behavior. Thoughts?